CMakeLists.txt

```diff
@@ -337,26 +337,27 @@ IF (ASPECT_PRECOMPILE_HEADERS)
   LIST(INSERT TARGET_SRC 0 ${HELPER_PATH})
 ENDIF()
 
-# Check whether we can find the WorldBuilder files. If so, make sure we
-# add a preprocessor #define that says that the WorldBuilder exists. Also 
-# add the new configure file to TARGET_SRC before DEAL_II_INVOKE_AUTOPILOT
-# is invoked.
-EXECUTE_PROCESS(COMMAND ls ${CMAKE_SOURCE_DIR}/contrib/WorldBuilder/VERSION
-                RESULT_VARIABLE result
-                OUTPUT_QUIET ERROR_QUIET)
-IF (result)
+# Check whether we can find the WorldBuilder and enable it.
+SET(ASPECT_WITH_WORLD_BUILDER ON CACHE BOOL "If ON the World Builder is compiled togheter with ASPECT.")
+IF (NOT EXISTS ${CMAKE_SOURCE_DIR}/contrib/WorldBuilder/VERSION)
   MESSAGE("-- World Builder not found.")
   SET(ASPECT_WITH_WORLD_BUILDER OFF CACHE BOOL "" FORCE)
-ELSE()
+ENDIF()
+IF (ASPECT_WITH_WORLD_BUILDER)
   SET(WORLD_BUILDER_SOURCE_DIR "${CMAKE_SOURCE_DIR}/contrib/WorldBuilder")
   INCLUDE("${WORLD_BUILDER_SOURCE_DIR}/cmake/version.cmake")
   MESSAGE("-- World Builder version ${WORLD_BUILDER_VERSION} found.")
-  SET(ASPECT_WITH_WORLD_BUILDER ON CACHE BOOL "If ON the World Builder is compiled togheter with ASPECT.")
 
-  configure_file("${WORLD_BUILDER_SOURCE_DIR}/source/config.cc.in" "${WORLD_BUILDER_SOURCE_DIR}/source/config.cc" @ONLY)
+  # add source and include dirs:
+  FILE(GLOB_RECURSE _files "contrib/WorldBuilder/source/*.cc")
+  LIST(APPEND TARGET_SRC ${_files})
+  INCLUDE_DIRECTORIES("${WORLD_BUILDER_SOURCE_DIR}/include/")
 
-  LIST(INSERT TARGET_SRC 0 "${WORLD_BUILDER_SOURCE_DIR}/source/config.cc")
+  # generate config.cc and include it:
+  CONFIGURE_FILE("${WORLD_BUILDER_SOURCE_DIR}/source/config.cc.in" "${CMAKE_BINARY_DIR}/world_builder_config.cc" @ONLY)
```

Comment:
This creates a problem for me (multiple definitions of version etc.), because I also use the world builder directory directly. Is it a problem for you to have the config file in "${WORLD_BUILDER_SOURCE_DIR}/source/config.cc"?

Reply:
You should not put temporary files into the source directory. If you have done so in the past, you will get linker errors (because you have two different .cc files) with this change. So I guess this means you also need to change the location in your CMakeLists.txt.
The reasoning is that an out-of-source build should not touch the source directory. Imagine you have two out-of-source builds. Now both will start overwriting or using the same file. Even if they will have the same content (for now at least unless other configuration options will be in that file!), configuring a new build directory will cause a rebuild of another build directory as the file is modified.
In summary: I think it is a bad design to modify/clutter the source directory.

Reply:
Hmm, that is a good point. I always viewed it as a source file, and that it therefore should be in the source directory. But you are right, that it makes more sense to put it in the build directory.

I have change the configuration on the and tested it and it seems to work fine now. Could you update the world builder, so that it links to a version which has separated out the config file? Otherwise it is good to merge.